### DISCUSSION TITLE
Netty Concurrent Request Issue

### DISCUSSION THREAD

**Discussion by satishver019** [netty/netty#11808] [Q&A]
Hi Team,

I am using netty TCP server to accept multiple remote TCP connections. I observed after few simultaneous parallel requests , netty TCP server doesn't respond

Do we have any parameter to configure in Netty TCP server  which is supposed to  accept multiple TCP connections as per the configured parameter ?

Right now, while bootstrapping the server, I have following setup

ServerBootstrap b = new ServerBootstrap();
        **b.group(new NioEventLoopGroup(1), new NioEventLoopGroup(20))**
                .channel(NioServerSocketChannel.class)
                .handler(new LoggingHandler(LogLevel.DEBUG))
                .childHandler(serverChannelInitializer);
        **b.option(ChannelOption.SO_BACKLOG, 100);**

Please suggest...

**Comment by normanmaurer** [comment]
There is nothing more to configure here... Attach a profiler and check if you maybe block the EventLoop.

**Reply by satishver019** [reply]
Thanks. I would like to understand the implication of below two cases

ServerBootstrap b = new ServerBootstrap();

1.  b.group(new NioEventLoopGroup(1), new NioEventLoopGroup(20)

**vs**

2. b.group(new NioEventLoopGroup())    //Does it handle concurrency in much smarter way than above point 1.

**Reply by normanmaurer** [reply]
The 1) will use a dedicated thread for just accepting connections while the 2) will share the accepting thread also with the accepted channels. Most of the time 2) is a good start

**Reply by satishver019** [reply]
Thanks Norman.
